### ISSUE TITLE
Prepare dataset

### ISSUE THREAD

**Issue by Ned09** [YadiraF/DECA#71]
Can you please give me more explanation about preparing data for training the model from scratch?
I want to use VGGFace2 for the first step. Do I need to generate landmarks for every image and save the landmarks as a text file or npy file?
and what about masks?
I really appreciate it if you give me some clue

I am trying to do this step:
b. Prepare label
FAN to predict 68 2D landmark
face_segmentation to get skin mask

**Comment by littlePrince126** [comment]
How do you get the five faces of every object?  (vggface2_val_list_max_normal_100_ring_5_1_serial.npy')
I could get landmarks or skin mask. Author choose five normoal faces with every identity. It is more important!

**Comment by Shelomith** [comment]
@littlePrince126 
Hi, what's your GPU used when runing face-segmentation? when I use caffe-gpu env with docker, 

F1103 05:56:53.555362   430 pooling_layer.cu:212] Check failed: error == cudaSuccess (8 vs. 0)  invalid device function

I got this error, It may because the difference of GPU computation ability between authur's and mine.

What's your situation?

**Comment by littlePrince126** [comment]
@Shelomith Telsa P100

**Comment by 1180800817** [comment]
> identity. It is more important!

Have you got the file(vggface2_val_list_max_normal_100_ring_5_1_serial.npy)? I would really appreciate it , if you share it with me.

**Comment by littlePrince126** [comment]
@1180800817 I generate it by myself. You can generate it with dataset（VggFace2，BUPT-Face，voxceleb2 ）.
voxceleb2 is a video dataset, you should choose good quality videos first, then get the images from videos.

**Comment by lhyfst** [comment]
@littlePrince126 
Can you reproduce the reported result on NoW using your generated data?
Additionally, could you please share vggface2_val_list_max_normal_100_ring_5_1_serial.npy with me? Even a demo is helpful!
I just want to know the explicit format of this file.

**Comment by littlePrince126** [comment]
@lhyfst I trained the model without any pretrained model. Finally, I get  median1.217mm  mean1.517mm std 1.249mm with NOW validation set. (DECA median:1.18, mean:1.46mm,std:1.25mm)

**Comment by Shelomith** [comment]
I've test DECA pretrained model in NoW dataset, result is median:1.15, mean:1.44mm,std:1.24mm... 
but in DECA paper they report median:1.09, mean:1.38mm,std:1.18mm 
@littlePrince126  朋友，方便加个微信讨论吗？

**Comment by Shelomith** [comment]
@littlePrince126 这我邮箱
shelomith1001@gmail.com

**Comment by littlePrince126** [comment]
@Shelomith Because your results is based on validation set. **(median:1.09, mean:1.38mm,std:1.18mm )**This result is based on test set. But I only get results(meadian:1.1778,Mean:1.4636,std:1.2529) with DECA model.

**Comment by littlePrince126** [comment]
> meadian:1.1778,Mean:1.4636,std:1.2529
This result is the same as  papr(DECA)

**Comment by Shelomith** [comment]
@littlePrince126 加个微信吧shelomith1001

**Comment by super3kl** [comment]
> I've test DECA pretrained model in NoW dataset, result is median:1.15, mean:1.44mm,std:1.24mm... but in DECA paper they report median:1.09, mean:1.38mm,std:1.18mm @littlePrince126 朋友，方便加个微信讨论吗？
hello 你好 最近也在跑这个代码 方便加个微信讨论吗？或者邮箱？

**Comment by Shelomith** [comment]
@super3kl wx: shelomith1001

**Comment by Victcode** [comment]
@littlePrince126 Hi,can you share the method of choosing five normoal faces?Thank you!

**Comment by WMXwmxwalter** [comment]
> > I've test DECA pretrained model in NoW dataset, result is median:1.15, mean:1.44mm,std:1.24mm... but in DECA paper they report median:1.09, mean:1.38mm,std:1.18mm @littlePrince126 朋友，方便加个微信讨论吗？
> > hello 你好 最近也在跑这个代码 方便加个微信讨论吗？或者邮箱？

hello 你好 最近也在跑这个代码 方便加个微信讨论吗？或者邮箱？

**Comment by WMXwmxwalter** [comment]
> @littlePrince126 Can you reproduce the reported result on NoW using your generated data? Additionally, could you please share vggface2_val_list_max_normal_100_ring_5_1_serial.npy with me? Even a demo is helpful! I just want to know the explicit format of this file.

Do you get the npy file?Can you share it with me?

**Comment by Eric3778** [comment]
@littlePrince126 你好，我最近在跑这个代码实验，我想请教一下你是怎么进行数据处理的，可以的话能否分享一下数据处理的代码，非常感谢！

**Comment by pfeducode** [comment]
> shelomith1001



> > I've test DECA pretrained model in NoW dataset, result is median:1.15, mean:1.44mm,std:1.24mm... but in DECA paper they report median:1.09, mean:1.38mm,std:1.18mm @littlePrince126 朋友，方便加个微信讨论吗？
> > hello 你好 最近也在跑这个代码 方便加个微信讨论吗？或者邮箱？

可以加个微信或者邮箱吗

**Comment by chen990627** [comment]
> @lhyfst I trained the model without any pretrained model. Finally, I get median1.217mm mean1.517mm std 1.249mm with NOW validation set. (DECA median:1.18, mean:1.46mm,std:1.25mm)

您好，方便给个联系方式吗？想请教一下怎么训练的，有偿。

**Comment by prokaryote-hgy** [comment]
@chen990627 @pfeducode @Eric3778 你们复现出来了吗？方便微信交流一下吗？

**Comment by prokaryote-hgy** [comment]
> choose good quality videos

Hi, how do you choose good quality videos? @littlePrince126

**Comment by YalanHe** [comment]
> I've test DECA pretrained model in NoW dataset, result is median:1.15, mean:1.44mm,std:1.24mm... but in DECA paper they report median:1.09, mean:1.38mm,std:1.18mm @littlePrince126 朋友，方便加个微信讨论吗？

你好，方便加一下微信一起讨论吗？

**Comment by YalanHe** [comment]
> > I've test DECA pretrained model in NoW dataset, result is median:1.15, mean:1.44mm,std:1.24mm... but in DECA paper they report median:1.09, mean:1.38mm,std:1.18mm @littlePrince126 朋友，方便加个微信讨论吗？
> > hello 你好 最近也在跑这个代码 方便加个微信讨论吗？或者邮箱？

你好，方便加一下微信一起讨论吗？

**Comment by YalanHe** [comment]
> @littlePrince126 你好，我最近在跑这个代码实验，我想请教一下你是怎么进行数据处理的，可以的话能否分享一下数据处理的代码，非常感谢！
你好，方便加一下微信一起讨论吗？

**Comment by YalanHe** [comment]
> > shelomith1001
> 
> > > I've test DECA pretrained model in NoW dataset, result is median:1.15, mean:1.44mm,std:1.24mm... but in DECA paper they report median:1.09, mean:1.38mm,std:1.18mm @littlePrince126 朋友，方便加个微信讨论吗？
> > > hello 你好 最近也在跑这个代码 方便加个微信讨论吗？或者邮箱？
> 
> 可以加个微信或者邮箱吗

你好，方便加一下微信一起讨论吗？

**Comment by YalanHe** [comment]
> > @lhyfst I trained the model without any pretrained model. Finally, I get median1.217mm mean1.517mm std 1.249mm with NOW validation set. (DECA median:1.18, mean:1.46mm,std:1.25mm)
> 
> 您好，方便给个联系方式吗？想请教一下怎么训练的，有偿。

你好，方便加一下微信一起讨论吗？

**Comment by YalanHe** [comment]
> @chen990627 @pfeducode @Eric3778 你们复现出来了吗？方便微信交流一下吗？

你好，方便加一下微信一起讨论吗？

**Comment by YalanHe** [comment]
> > @chen990627 @pfeducode @Eric3778 你们复现出来了吗？方便微信交流一下吗？
> 
> 你好，方便加一下微信一起讨论吗？
wx是wxid_3y4z9tfwyi8b22

**Comment by prokaryote-hgy** [comment]
@YalanHe 你好，查找不到该微信账号，是否账号有误

**Comment by YalanHe** [comment]
> @YalanHe 你好，查找不到该微信账号，是否账号有误

你好，试一下手机号可以吗？15216727179

**Comment by nainsheng** [comment]
> 你能帮我制作面膜吗？我无法安装 https://github.com/YuvalNirkin/face_segmentation
> 
> 当我运行存储库时，出现此错误 ![image](https://user-images.githubusercontent.com/92327180/138352841-1af5fed2-ee79-4499-b0b5-154cfdae6f93.png)

May I ask how you resolved this issue and ultimately achieved training?

**Comment by mumucc1** [comment]
> How do you get the five faces of every subject? (vggface2_val_list_max_normal_100_ring_5_1_serial.npy')
> I could get landmarks or skin mask. Author choose five normoal faces with every identity. It is more important!

Hi！Can you reproduce the model training？
